guides/source/configuring.md

```diff
@@ -3560,6 +3560,69 @@ development:
   query_cache: false
 ```
 
+### Database Pooling
+
+Active Record database connections are managed by [`ActiveRecord::ConnectionAdapters::ConnectionPool`][] which ensures that a connection pool synchronizes the amount of thread access to a limited number of database connections. Connection pools can be configured in `database.yml`.
+
+```yaml
+development:
+  adapter: sqlite3
+  database: storage/development.sqlite3
+  idle_timeout: 300
+  keepalive: 600
+  max_age: 600
+  max_connections: 5
+  min_connections: 0
+  timeout: 5000
+```
+
+Since the connection pool is managed by Active Record, all application servers
+(Thin, Puma, Unicorn, etc.) should behave the same. The database connection
+pool is initially empty. As demand for connections increases it will create
+them until it reaches the limit defined by `max_connections`. The demand for
+connections will generally be constrained by the number of application threads.
+
+A request will check out a connection when it requires access to the database.
+After completing the query, the connection will be checked back in, and thus
+available to be reused.
+
+Especially when an application has multiple databases, creating new connections
+on demand can be expensive. When `min_connections` is set, Active Record can
+eagerly open and maintain a minimum number of database connections.
+
+Database connections will be recycled/retired in order of creation. Connections
+are retired when their age exceeds `max_age`, or they have been idle for longer
+than `idle_timeout`.
+
+<!-- Suggestions please -->
+Idle database connections can be bad - the database may no longer exist, or
+may assume the application server no longer exists. `keepalive` will check
+open database connections and keep them alive.
+
+<!-- How does this interact with idle_timeoue? -->
+<!-- How does this interact with min_connections? -->
+<!-- How does this interact with INFINITY? -->
```

Comment:
cc @matthewd 

Hi, could you kindly offer some suggestions or thoughts on how these configurations interact with each other?

Maybe some of these details could/should be moved to https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/ConnectionPool.html ?